File: backend/src/main/java/com/yourcompany/onlineexam/model/Question.java
package com.yourcompany.onlineexam.model;

import java.util.List;

public class Question {
    private String id;
    private String content;
    private String type; // "true_false", "multiple_choice", "essay"
    private String level; // "easy", "medium", "hard"
    private List<Option> options; // Sửa lại kiểu dữ liệu
    private String answer; // Đáp án đúng
    private String questionBankId; // Thuộc ngân hàng đề nào

    public Question() {}

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }

    public String getType() { return type; }
    public void setType(String type) { this.type = type; }

    public String getLevel() { return level; }
    public void setLevel(String level) { this.level = level; }

    public List<Option> getOptions() { return options; }
    public void setOptions(List<Option> options) { this.options = options; }

    public String getAnswer() { return answer; }
    public void setAnswer(String answer) { this.answer = answer; }

    public String getQuestionBankId() { return questionBankId; }
    public void setQuestionBankId(String questionBankId) { this.questionBankId = questionBankId; }

    // Inner class Option
    public static class Option {
        private String id; // Thêm id cho option
        private String text;
        private boolean correct;

        public Option() {}

        public String getId() { return id; }
        public void setId(String id) { this.id = id; }
        public String getText() { return text; }
        public void setText(String text) { this.text = text; }
        public boolean isCorrect() { return correct; }
        public void setCorrect(boolean correct) { this.correct = correct; }
    }
}

File: backend/OnlineExam/Models/Question.cs
using Google.Cloud.Firestore;

namespace OnlineExam.Models;

[FirestoreData]
public class Question
{
    [FirestoreDocumentId]
    public string? Id { get; set; }
    [FirestoreProperty("content")]
    public string? Content { get; set; }
    [FirestoreProperty("type")]
    public string? Type { get; set; }
    [FirestoreProperty("level")]
    public string? Level { get; set; }
    [FirestoreProperty("options")]
    public List<QuestionOption>? Options { get; set; }
    [FirestoreProperty("answer")]
    public string? Answer { get; set; }
    [FirestoreProperty("questionBankId")]
    public string? QuestionBankId { get; set; }
}

[FirestoreData]
public class QuestionOption
{
    [FirestoreProperty("id")]
    public string? Id { get; set; }
    [FirestoreProperty("text")]
    public string? Text { get; set; }
    [FirestoreProperty("correct")]
    public bool Correct { get; set; }
}
